### DISCUSSION TITLE
Recommended approach for spawning subprocess in browser

### DISCUSSION THREAD

**Discussion by bryanchenmchp** [eclipse-theia/theia#13021] [Q&A]
Hi, I am currently developing an IDE that uses Theia.

Our IDE has a couple tasks that require running short, one-off cli commands in the background from the browser / electron process. Is there an API call that achieves this, otherwise what will I need to write myself?

* The [task](https://github.com/eclipse-theia/theia/tree/master/packages/task) package can be used to run cli commands, but requires attaching the task to a task terminal widget, which is something that I'm trying to avoid.
* The [process](https://github.com/eclipse-theia/theia/tree/master/packages/process) package seems to have an API related to spawning subprocesses, but only provides code for node instead of the browser.

**Comment by msujew** [comment]
Hey @bryanchenmchp,

Since you're trying to avoid attaching a terminal to your process, the best way would be to implement something via our RPC protocol and call that from the frontend. That way you have full control over what process you start and what you do with its output. See [our documentation](https://theia-ide.org/docs/json_rpc/) on how to do that.

**Reply by bryanchenmchp** [reply]
Mmm I see, so essentially a frontend RPC wrapper over [process](https://github.com/eclipse-theia/theia/tree/master/packages/process)?
